### ISSUE TITLE
Template Method

### ISSUE THREAD

**Issue by dgruntz** [iluwatar/java-design-patterns#1099]
As statet in the referenced tutorial [Template-method Pattern Tutorial](https://www.journaldev.com/1763/template-method-design-pattern-in-java)
I would add a note in the pattern description that the template method should be declared `final`. 

> To make sure that subclasses don’t override the template method, we should make it final.

And I would change the code accordingly.

**Comment by iluwatar** [comment]
Yes, I agree with you.

**Comment by liuc-ai** [comment]
That‘s right.

**Comment by JuanfPuebla** [comment]
Asi es.

**Comment by elouie-code** [comment]
@iluwatar can you please assign this to me?

**Comment by elouie-code** [comment]
@iluwatar @dgruntz 
where would you want to put this new comment?
`To make sure that subclasses don’t override the template method, we should make it final.`

**Comment by dgruntz** [comment]
I do not want to add this comment but rather change the code and make the template method final. This change has already been made with PR #1100, but as long as this PR is open this issue remains open as well; and the PR can't be merged yet as there is a [issue](https://github.com/mockito/mockito/issues/1826) in Mockito which has not yet been fixed.

**Comment by iluwatar** [comment]
@elouie-code it's assigned to you now

**Comment by ohbus** [comment]
@elouie-code is there any update regarding this issue or any progress that you made?

**Comment by syfds** [comment]
hi @ohbus , @iluwatar , looks like nothing happens here, can I do a PR for this ticket?

**Comment by dgruntz** [comment]
The PR #1100 is already open, but the PR can't be merged yet as there is a [issue](https://github.com/mockito/mockito/issues/1826) in Mockito which has not yet been fixed. I just checked whether the problem has been fixed with version 3.12.4 of Mockito, but this is not the case.

**Comment by syfds** [comment]
I see, but in the PR #1100 there is no adjustment of README documentation (explanation why template method should be marked "final" and changing the sample code).

**Comment by dgruntz** [comment]
Agreed, this could be added, and probably the class diagram needs to be adjusted as well (I don't know whether final methods are specially marked, in UML I would use a <<final>> stereotype).

**Comment by syfds** [comment]
ok, so assign this ticket to me if you want to have the adjustment of the documentation:
* explanation why final is important in the Template Pattern
* adjustment of all code snippets
* adjustment of the class diagram

@dgruntz to be honest I wouldn't wait with your change until mockito fixes the issue with final methods. I would merge the code (without unit test) and create a reminder ticket for future to create a unit test with mockito so you are not blocked with the change in the sample. What do you think about it?

**Comment by ohbus** [comment]
@syfds We could do that. We can merge the code without the unit tests which would drop code coverage and is not a good thing to do, though these are not production codes.

We can wait for @iluwatar's suggestion on this. What do you suggest @dgruntz?

**Comment by iluwatar** [comment]
What @syfds suggested sounds reasonable to me

**Comment by shibicr93** [comment]
Hello, @iluwatar, I would like to work on this please assign this issue to me

**Comment by iluwatar** [comment]
All right, good luck @shibicr93

**Comment by dgruntz** [comment]
Fixed with PR #2057

**Comment by shibicr93** [comment]
@iluwatar  please assign this to @dgruntz  since he has already fixed it

**Comment by iluwatar** [comment]
Let's keep the issue open until the PR has been merged
